core/arch/arm/kernel/thread.c

```diff
@@ -957,6 +958,43 @@ static uint32_t __maybe_unused get_midr_primary_part(uint32_t midr)
 	       MIDR_PRIMARY_PART_NUM_MASK;
 }
 
+#ifdef ARM64
+static bool probe_workaround_available(void)
+{
+	int32_t r;
+
+	r = thread_smc(SMCCC_VERSION, 0);
+	if (r < 0)
+		return false;
+	if (r < 0x10001)	/* compare with version 1.1 */
+		return false;
+
+	/* Version >= 1.1, so SMCCC_ARCH_FEATURES is available */
+	r = thread_smc(SMCCC_ARCH_FEATURES, SMCCC_ARCH_WORKAROUND_1);
+	return r >= 0;
+}
+
+static vaddr_t select_vector(vaddr_t a)
+{
+	if (probe_workaround_available()) {
+		DMSG("SMCCC_ARCH_WORKAROUND_1 (%#08" PRIx32 ") available",
+		     SMCCC_ARCH_WORKAROUND_1);
+		DMSG("Workaround for CVE-2017-5715 used");
+		return a;
+	}
+
+	DMSG("SMCCC_ARCH_WORKAROUND_1 (%#08" PRIx32 ") unavailable",
+	     SMCCC_ARCH_WORKAROUND_1);
+	DMSG("Workaround for CVE-2017-5715 not needed (if ARM-TF is up to date)");
```

Comment:
I don't understand this. You're saying that it is enough to have an up-to-date ARM-TF to mitigate the CVE? How can this protect the EL0/EL1 transition? ARM-TF is not involved in this path, is it?

Reply:
If the workaround isn't needed (fixed in newer cores or whatever) ARM-TF doesn't implement the workaround. Provided that ARM-TF is up to date.

Reply:
Ah, OK. Now this makes sense.